jvmhs/src/Jvmhs/Data/Code.hs

```diff
@@ -158,9 +165,423 @@ verificationTypeInfo g (B.StackMapTable s) =
+
+
+instance ToJSON (B.AbsVariableMethodId B.High) where
+  toJSON = \case
+    B.VInterfaceMethodId interface_method -> object
+      [ "kind" .= String "interface_method"
+      , "method" .= toJSON interface_method
+      ]
+    B.VMethodId (B.InClass a b) -> object
+      [ "kind" .= String "method"
+      , "class" .= B.classNameAsText a
+      , "method" .= methodIDToText b
+      ]
+
+
+instance ToJSON (B.AbsInterfaceMethodId B.High) where
+  toJSON interfacemethod = object
+      [ 
+        "type" .= String "abs_interface_method"
+      , "method" .= toJSON interfacemethod
+      ]
+
+
+instance ToJSON (B.InvokeDynamic B.High) where
+  toJSON (B.InvokeDynamic attrIndex method) = object
+      [ 
+        "attr_index" .= show (attrIndex)
+      , "method" .= toJSON method
+      ]
+
+instance ToJSON (B.MethodId) where
+  toJSON (B.MethodId (B.NameAndType name args)) = object
+      [ 
+       "signature" .= Text.concat [name,":",(B.toText args)]
+      ]
+
+methodIDToText :: B.MethodId -> Text.Text
+methodIDToText (B.MethodId (B.NameAndType name args)) = Text.concat [name,":",(B.toText args)]
+
+
+nameAndTypeFromFieldId :: B.FieldId -> Text.Text
+nameAndTypeFromFieldId (B.FieldId (B.NameAndType name type_)) = Text.concat [name,":",B.toText type_] 
+
+
+getListOfPairsFromBConstant = \case
+    Just (VInteger a) ->
+      [ "type" .= String "I"
+      , "value" .= toJSON a
+      ]
+    Just (VFloat a) ->
+      [ "type" .= String "F"
+      , "value" .= toJSON a
+      ]
+    Just (VDouble a) ->
+      [ "type" .= String "D"
+      , "value" .= toJSON a
+      ]
+    Just (VString a) ->
+      [ "type" .= String "string"
+      , "value" .= toJSON a
+      ]
+    Just (VClass a) ->
+      [ "type" .= String "class"
+      , "value" .= toJSON a
+      ]
+    Just (VMethodType a) ->
+      [ "type" .= String "method"
+      , "value" .= toJSON a
+      ]
+    Just (VMethodHandle a) ->
+      [ "type" .= String "method_handle"
+      , "value" .= toJSON a
+      ]
+    Nothing ->
+      [ "type" .= String "null"
+      ]
+
+getInvocationAttributes = \case
+    B.InvkSpecial (B.VInterfaceMethodId a) ->
+      [ "kind" .= String "special"
+      , "ref" .= toJSON a
+      ]
+    B.InvkSpecial (B.VMethodId (B.InClass a b)) ->
+      [ "kind" .= String "special" 
+      , "class" .= B.classNameAsText a
+      , "method" .= methodIDToText b
+      ]
+    B.InvkVirtual (B.InClass a b) ->
+      [ "kind" .= String "virtual"
+      , "class" .= B.classNameAsText a
+      , "method" .= methodIDToText b
+      ]  
+    B.InvkStatic (B.VInterfaceMethodId a) ->
+      [ "kind" .= String "static"
+      , "ref" .= toJSON a
+      ]
+    B.InvkStatic (B.VMethodId (B.InClass a b)) ->
+      [ "kind" .= String "static"
+      , "class" .= B.classNameAsText a
+      , "method" .= methodIDToText b
+      ]
+    B.InvkInterface word8 (B.AbsInterfaceMethodId (B.InClass a b))->
+      [ "kind" .= String "interface"
+      , "word8" .= toJSON word8
```

Comment:
Give propper name.

Reply:
Not sure of the interpretation of this. There are a couple of other places I have done the same thing.